futurex_openedx_extensions/dashboard/serializers.py

```diff
@@ -921,3 +922,22 @@ class AggregatedCountsSerializer(ReadOnlySerializer):
     all_tenants = AggregatedCountsAllTenantsSerializer()
     by_tenant = AggregatedCountsOneTenantSerializer(many=True)
     limited_access = serializers.BooleanField()
+
+
+class FileUploadSerializer(ReadOnlySerializer):
+    """
+    Serializer for handling the file upload request. It validates and serializes the input data.
+    """
+    file = serializers.FileField()
+    slug = serializers.CharField(max_length=255)
+    tenant_id = serializers.IntegerField()
+
+    def validate_tenant_id(self, value: int) -> int:  # pylint:disable=no-self-use
+        """
+        Custom validation for tenant_id to ensure that the tenant exists.
+        """
+        try:
+            TenantConfig.objects.get(id=value)
+        except TenantConfig.DoesNotExist as exc:
+            raise serializers.ValidationError(f'Tenant with ID {value} does not exist.') from exc
+        return value
```

Comment:
we need to check for permissions, not just tenant existance
```suggestion
        if not self.context.get('request') or not hasattr(self.context.get('request'), 'fx_permission_info'): ...error...

        if tenant_id not in self.context.get('request').fx_permission_info['view_allowed_tenant_ids_any_access']:...
```

Reply:
Updated this one, I'll check in other API's and will open a new PR for those fixes.

Reply:
yes please, update them all as needed